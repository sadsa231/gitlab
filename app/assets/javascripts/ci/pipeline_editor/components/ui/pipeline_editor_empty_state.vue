<script>
import { GlButton, GlSprintf } from '@gitlab/ui';
import { __ } from '~/locale';
import PipelineEditorFileNav from '~/ci/pipeline_editor/components/file_nav/pipeline_editor_file_nav.vue';

export default {
  components: {
    GlButton,
    GlSprintf,
    PipelineEditorFileNav,
  },
  i18n: {
    title: __('Optimize your workflow with CI/CD Pipelines'),
    body: __(
      'Create a new %{codeStart}.gitlab-ci.yml%{codeEnd} file at the root of the repository to get started.',
    ),
    btnText: __('Configure pipeline'),
    externalCiNote: __("This project's pipeline configuration is located outside this repository"),
    externalCiInstructions: __(
      'To edit the pipeline configuration, you must go to the project or external site that hosts the file.',
    ),
  },
  inject: {
    emptyStateIllustrationPath: {
      default: '',
    },
    usesExternalConfig: {
      default: false,
      type: Boolean,
      required: false,
    },
  },
  methods: {
    createEmptyConfigFile() {
      this.$emit('createEmptyConfigFile');
    },
  },
};
</script>
<template>
  <div>
    <pipeline-editor-file-nav v-on="$listeners" />
    <div class="gl-display-flex gl-flex-direction-column gl-align-items-center gl-mt-11">
      <img :src="emptyStateIllustrationPath" />
      <div
        v-if="usesExternalConfig"
        class="gl-display-flex gl-flex-direction-column gl-align-items-center"
      >
        <h1 class="gl-font-size-h1">{{ $options.i18n.externalCiNote }}</h1>
        <p class="gl-mt-3">{{ $options.i18n.externalCiInstructions }}</p>
      </div>
      <div v-else class="gl-display-flex gl-flex-direction-column gl-align-items-center">
        <h1 class="gl-font-size-h1">{{ $options.i18n.title }}</h1>
        <p class="gl-mt-3">
          <gl-sprintf :message="$options.i18n.body">
            <template #code="{ content }">
              <code>{{ content }}</code>
            </template>
          </gl-sprintf>
        </p>
        <gl-button
          variant="confirm"
          class="gl-mt-3"
          data-qa-selector="create_new_ci_button"
          @click="createEmptyConfigFile"
        >
          {{ $options.i18n.btnText }}
        </gl-button>
      </div>
    </div>
  </div>
</template>
