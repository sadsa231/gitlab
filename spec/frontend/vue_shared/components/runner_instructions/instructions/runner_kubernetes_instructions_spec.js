import { shallowMount } from '@vue/test-utils';

import { GlButton } from '@gitlab/ui';
import RunnerKubernetesInstructions from '~/vue_shared/components/runner_instructions/instructions/runner_kubernetes_instructions.vue';

describe('RunnerKubernetesInstructions', () => {
  let wrapper;

  const createComponent = () => {
    wrapper = shallowMount(RunnerKubernetesInstructions, {});
  };

  const findButton = () => wrapper.findComponent(GlButton);

  beforeEach(() => {
    createComponent();
  });

  it('renders contents', () => {
    expect(wrapper.text().replace(/\s+/g, ' ')).toMatchSnapshot();
  });

  it('renders link', () => {
    expect(findButton().attributes('href')).toBe(
      'https://docs.gitlab.com/runner/install/kubernetes.html',
    );
  });
});
