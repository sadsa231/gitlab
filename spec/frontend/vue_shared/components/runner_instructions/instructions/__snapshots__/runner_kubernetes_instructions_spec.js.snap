// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RunnerKubernetesInstructions renders contents 1`] = `"To install Runner in Kubernetes follow the instructions described in the GitLab documentation. View installation instructions Close"`;
